### PR TITLE
Adds readme and yaml on details page

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
- This patch adds:
       - Implementation of readme and yaml on details page by making
         api calls to github using raw url of the resource
       - Renders readme and yaml using `react-markdown` and `syantax-highlighter`

Signed-off-by: Puneet Punamiya <ppunamiy@redhat.com>

* Screenshot of readme of a resource

image

* Screenshot of yaml of a resource

image

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: ui/src/containers/Details/Details.test.tsx [hunk 1]
```diff
@@ -6,6 +6,8 @@ import { FakeHub } from '../../api/testutil';
 import { createProviderAndStore } from '../../store/root';
 import Details from '.';
 import BasicDetails from '../BasicDetails';
+import Description from '../../components/Description';
+import ReactMarkdown from 'react-markdown';
```
Comment (vinamra28) [inline ui/src/containers/Details/Details.test.tsx:10]: organize imports please

File: ui/src/store/resource.test.ts [hunk 2]
```diff
@@ -523,4 +523,64 @@ describe('Store functions', () => {
...
+        categories: CategoryStore.create({}, { api })
+      }
+    );
+
+    expect(store.isLoading).toBe(true);
+
+    when(
+      () => !store.isLoading,
+      () => {
+        expect(store.resources.size).toBe(7);
```
Comment (pratap0007) [inline ui/src/store/resource.test.ts:541]: nit : does't need ,I guess

File: ui/src/components/Description/index.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+interface Props {
+  name: string;
+}
+
+const Description: React.FC<Props> = (props) => {
+  const { resources } = useMst();
+
+  const [activeTabKey, setActiveTabKey] = React.useState(0);
+  const handleTabClick = (
+    event: React.MouseEvent<HTMLElement, MouseEvent>,
```
[the remark this answers is not in the document]
Comment (PuneetPunamiya) [reply ui/src/components/Description/index.tsx:19]: Yes kept it is `_` but what would be the type of it :thinking:  ?

File: ui/src/containers/Details/index.tsx [hunk 4]
```diff
@@ -4,13 +4,16 @@ import { useParams } from 'react-router-dom';
...
 import BasicDetails from '../BasicDetails';
+import Description from '../../components/Description';
 
 const Details: React.FC = () => {
   const { resources } = useMst();
   const { name } = useParams();
 
   const updateVersions = () => {
     resources.versionInfo(name);
+    resources.readme(name);
```
Comment (pratap0007) [inline ui/src/containers/Details/index.tsx:15]: can you put in a separate function or rename the function `updateVersions` as general

File: ui/src/containers/Details/Details.test.tsx [hunk 5]
```diff
@@ -21,65 +23,96 @@ jest.mock('react-router-dom', () => {
...
+
+    const { resources } = root;
+    when(
+      () => {
+        return !resources.isLoading;
+      },
+      () => {
+        setTimeout(() => {
+          const resource = resources.filteredResources;
+          expect(resource.length).toBe(7);
```
Comment (pratap0007) [inline ui/src/containers/Details/Details.test.tsx:103]: does it need  this test?

**Review by vinamra28 — APPROVED** [review]
/lgtm
/woof

**Review by vinamra28 — COMMENTED** [review]
/lgtm

**Pull request merged**